internal/app/component/component_handler_events.go

```diff
@@ -69,3 +74,65 @@ type GetComponentIssueSeverityCountsEvent struct {
 func (e *GetComponentIssueSeverityCountsEvent) Name() event.EventName {
 	return GetComponentIssueSeverityCountsEventName
 }
+
+// OnComponentCreateAuthz is a handler for the CreateComponentEvent
+// It creates an OpenFGA relation tuple for the component and the current user
+func OnComponentCreateAuthz(db database.Database, e event.Event, authz openfga.Authorization) {
+	defaultPrio := db.GetDefaultIssuePriority()
+	defaultRepoName := db.GetDefaultRepositoryName()
+
+	l := logrus.WithFields(logrus.Fields{
+		"event":             "OnComponentCreateAuthz",
+		"payload":           e,
+		"default_priority":  defaultPrio,
+		"default_repo_name": defaultRepoName,
+	})
+
+	if createEvent, ok := e.(*CreateComponentEvent); ok {
+		componentId := strconv.FormatInt(createEvent.Component.Id, 10)
+		userId := authz.GetCurrentUser()
+
+		rlist := []openfga.RelationInput{
+			{
+				UserType:   "role",
+				UserId:     openfga.UserId(userId),
+				Relation:   "role",
+				ObjectType: "component",
+				ObjectId:   openfga.ObjectId(componentId),
+			},
+		}
+
+		for _, rel := range rlist {
+			authz.AddRelation(rel)
+		}
+	} else {
+		l.Error("Wrong event")
```

Comment:
Please use better error messages. Have a look at our [error packge](https://github.com/cloudoperators/heureka/tree/main/internal/errors). Have a look at this [example](https://github.com/cloudoperators/heureka/blob/main/internal/app/component_instance/component_instance_handler.go#L78-L81) how to use it.